src/server/frontend_xwayland/xwayland_surface.cpp

```diff
@@ -709,6 +715,13 @@ void mf::XWaylandSurface::attach_wl_surface(WlSurface* wl_surface)
         reply_function();
     }
 
+    if (rejected)
+    {
+        scene_surface_close_requested();
```

Comment:
The main job of `scene_surface_close_requested()` is to detect and use the `WM_DELETE_WINDOW` protocol, which allows clients to receive close requests (for example so they can show a "save your work?" popup instead of just being killed). Since the window is not being opened and closing is not optional it seems to me this shouldn't be used.

Perhaps this call can be removed entirely? Or perhaps replaced with a call to `xcb_kill_client()` (as `scene_surface_close_requested()` does when `WM_DELETE_WINDOW` isn't supported)?

I haven't tested it, but my guess would be the most correct thing to do would be to call `xcb_kill_client()` and then `connection->flush()` after the call to `close()`

Reply:
Agree this needs testing to confirm, but does sound like the right approach